packages/cosmo/package.py

```diff
@@ -54,6 +54,7 @@ class Cosmo(MakefilePackage):
     variant('slave', default='tsa', description='Build on slave tsa or daint', multi=False)
     variant('eccodes', default=False, description='Build with eccodes instead of grib-api')
 
+    conflicts('+cppdycore', when='%pgi cosmo_target=cpu')
```

Comment:
actually I wanted to say something like 

`when=' not %gcc cosmo_target=cpu')` 
or even better
`when=' not <dycore_compiler> cosmo_target=cpu')` 
but not idea how to write that. @elsagermann do you know?